framework/interactive/internal/interactive.cpp

```
@@ -354,11 +358,18 @@ async::Promise<io::path_t> Interactive::selectOpeningFile(const std::string& tit
 {
 #ifndef Q_OS_LINUX
     return async::make_promise<io::path_t>([title, dir, filter](auto resolve, auto reject) {
+#ifdef Q_OS_MAC
+        auto scope = std::make_shared<MacOSInteractiveHelper::NativeDialogScope>();
+#endif
         QFileDialog* dlg = new QFileDialog(nullptr, QString::fromStdString(title), dir.toQString(), filterToString(filter));
```

Clang (14.0.6)

[warning] variable 'dlg' is not initialized

(cppcoreguidelines-init-variables)